Modules/Core/SpatialObjects/test/itkPolygonSpatialObjectIsInsideInObjectSpaceTest.cxx

```diff
@@ -0,0 +1,143 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkPolygonSpatialObject.h"
+#include "itkGroupSpatialObject.h"
+#include "itkSpatialObjectToImageFilter.h"
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkTestingMacros.h"
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+// -------------------------------------------------------------------------------------
+// The purpose of this test is to make sure that the function
+// ObjectIsInsideInObjectSpace is working as expected.
+// This is related to a regression that was fixed under case 1082
+// which showed that certain contours that were created from the polygonSpatialObject
+// had trailing tails that were not supposed to be there.
+// In order to test this we take a known regression list of contour points
+// and create a contour image from these points.
+// We compare the resulting image with the known baseline contour image.
+// ---------------------------------------------------------------------------------------
+
+int
+itkPolygonSpatialObjectIsInsideInObjectSpaceTest(int argc, char * argv[])
+{
+  constexpr int Dimension = 2;
+  using PointType = itk::ImageBase<Dimension>::PointType;
+  using PolygonPointType = itk::PolygonSpatialObject<Dimension>::PolygonPointType;
+  using PolygonPointListType = itk::PolygonSpatialObject<Dimension>::PolygonPointListType;
+  using PixelType = uint16_t;
+  using Group = itk::GroupSpatialObject<Dimension>;
+  using ImageType = itk::Image<PixelType, Dimension>;
+  using SpatialObjectToImageFilterType = itk::SpatialObjectToImageFilter<Group, ImageType>;
+
+  if (argc < 3)
+  {
+    std::cerr << "Missing Parameters" << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv);
+    std::cerr << " polygonPointsCSVFile baselineImageFileName outImageFileName ";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  const char * csvFileName = argv[1];
+  const char * baselineFileName = argv[2];
+  const char * outFileName = argv[3];
+
+  try
```

Comment:
Please avoid using large `try/catch` blocks: if they are necessary, they should apply to the few ITK statements that are likely to cause exceptions (e.g. the declaration `PolygonPointListType polygonPointList;` is unlikely to cause an exception). In tests, we encourage using the `ITK_TRY_EXPECT_NO_EXCEPTION` macro, which completely avoids any boilerplate code, e.g.
```
try
{
  myFilter->Update():
}
catch(itk::ExceptionObject & exception)
{
    std::cerr << "Exception: ";
    std::cerr << exception.what() << std::endl;
    return EXIT_FAILURE;
}
```

becomes
```
ITK_TRY_EXPECT_NO_EXCEPTION(myFilter->Update);
```

Reply:
Solved.